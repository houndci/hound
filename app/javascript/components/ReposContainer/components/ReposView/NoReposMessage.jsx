import React from 'react';

const NoReposMessage = () => (
  <div className="no-repos-message">
    You haven't added any GitHub repos yet.
  </div>
);

export default NoReposMessage;
