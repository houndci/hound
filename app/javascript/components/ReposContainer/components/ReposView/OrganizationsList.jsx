import React from 'react';
import _ from 'lodash';

import Organization from '../Organization';

export default class OrganizationsList extends React.Component {
  reposForOrg(org) {
    if (_.has(org, "id")) {
      return _.filter(this.props.repos, (repo) => {
        return repo.owner.id === org.id;
      });
    } else {
      return _.filter(this.props.repos, (repo) => {
        return this.orgName(repo.name) === org.name;
      });
    }
  }

  orgName(name) {
    return _.split(name, "/")[0];
  }

  render() {
    const {
      repos,
      onRepoClicked,
      filterTerm,
      isProcessingId,
      organizations,
    } = this.props;

    return (
      <section className="organizations">
        {organizations.map( org => (
          <Organization
            name={org.name}
            key={org.id || org.name}
            repos={this.reposForOrg(org)}
            onRepoClicked={onRepoClicked}
            filterTerm={filterTerm}
            isProcessingId={isProcessingId}
            configEnabled={org.config_enabled}
            configRepo={org.config_repo}
            ownerId={org.id}
          />
        ))}
      </section>
    );
  }
}
