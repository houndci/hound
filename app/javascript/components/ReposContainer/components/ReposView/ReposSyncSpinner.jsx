import React from 'react';

export default class ReposSyncSpinner extends React.Component {
  render() {
    return (
      <div className="repos-syncing">
        <div className="dot"></div>
        <div className="dot"></div>
        <div className="dot"></div>
      </div>
    );
  }
}
