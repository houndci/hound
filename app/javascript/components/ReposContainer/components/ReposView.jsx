import React from 'react';

import ReposSyncSpinner from './ReposView/ReposSyncSpinner';
import OrganizationsList from './ReposView/OrganizationsList';

export default class ReposView extends React.Component {
  render() {
    const {
      isSyncing,
      organizations,
      repos,
      filterTerm,
      onRepoClicked,
      isProcessingId
    } = this.props;

    if (isSyncing) {
      return (
        <ReposSyncSpinner/>
      );
    } else {
      return (
        <OrganizationsList
          organizations={organizations}
          repos={repos}
          filterTerm={filterTerm}
          onRepoClicked={onRepoClicked}
          isProcessingId={isProcessingId}
        />
      );
    }
  }
}
