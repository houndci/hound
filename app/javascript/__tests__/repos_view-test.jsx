import ReposView from '../components/ReposContainer/components/ReposView';

it('renders appropriately', () => {
  const organizations = [
    { id: 1, name: "Test org" }
  ];
  const repos = [
    {
      id: 1,
      name: "Test repo",
      owner: {
        id: 1
      }
    }
  ];

  const wrapper = shallow(
    <ReposView
      isSyncing={false}
      organizations={organizations}
      repos={repos}
      filterTerm={""}
      onRepoClicked={(event) => onRepoClicked}
      isProcessingId={null}
     />
  );
  expect(wrapper).toMatchSnapshot();
});
