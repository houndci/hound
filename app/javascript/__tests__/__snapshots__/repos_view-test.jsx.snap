// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders appropriately 1`] = `
<OrganizationsList
  filterTerm=""
  isProcessingId={null}
  onRepoClicked={[Function]}
  organizations={
    Array [
      Object {
        "id": 1,
        "name": "Test org",
      },
    ]
  }
  repos={
    Array [
      Object {
        "id": 1,
        "name": "Test repo",
        "owner": Object {
          "id": 1,
        },
      },
    ]
  }
/>
`;
